core/application/notificationWorkflow/NotificationService.ts

```diff
@@ -1,54 +1,122 @@
+        `Failed to create notification. Error: ${error}`,
+        GENERIC_CODES.ERROR
+      )
+    }
+  }
 
+  async createBloodDonationNotification(
+    notificationAttributes: BloodDonationNotificationAttributes,
+    notificationRepository: NotificationRepository<BloodDonationNotificationDTO>
+  ): Promise<void> {
+    try {
+      const { userId, type, payload } = notificationAttributes
       if (payload !== undefined && ['BLOOD_REQ_POST', 'REQ_ACCEPTED'].includes(type)) {
-        const requestPostAttributes = notificationAttributes as BloodPostNotificationAttributes
-        const existingItem = await notificationRepository.getItem(
-          `NOTIFICATION#${userId}`,
-          `${type}#${payload.requestPostId}`
+        const requestPostAttributes = notificationAttributes
+        const existingItem = await notificationRepository.getBloodDonationNotification(
+          userId,
+          requestPostAttributes.payload.requestPostId,
+          type
         )
 
-        if (existingItem !== null) {
-          return 'Already notified'
-        } else {
+        if (existingItem === null) {
           await notificationRepository.create({
             ...notificationAttributes,
             status: NotificationStatus.PENDING,
             id: requestPostAttributes.payload.requestPostId,
             createdAt: new Date().toISOString()
           })
         }
-      } else {
-        await notificationRepository.create({
-          ...notificationAttributes,
-          status: NotificationStatus.PENDING,
-          id: generateUniqueID(),
-          createdAt: new Date().toISOString()
-        })
+      }
+    } catch (error) {
+      throw new NotificationOperationError(
+        `Failed to create notification. Error: ${error}`,
+        GENERIC_CODES.ERROR
+      )
+    }
+  }
+
+  async updateBloodDonationNotifications(
+    requestPostId: string,
+    notificationPayload: Partial<BloodDonationPayloadAttributes>,
+    notificationRepository: NotificationRepository<BloodDonationNotificationDTO>
+  ): Promise<string> {
+    try {
+      const existingNotifications = await notificationRepository.queryBloodDonationNotifications(
+        requestPostId
+      )
+
+      if (existingNotifications === null) {
+        throw new NotificationOperationError(
+          'Notifications does not exist.',
+          GENERIC_CODES.NOT_FOUND
+        )
       }
 
-      await snsModel.publish(notificationAttributes, userSnsEndpointArn)
-      return 'Notified user successfully.'
+      for (const notification of existingNotifications) {
+        const updatedNotification: Partial<BloodDonationNotificationDTO> = {
+          id: requestPostId,
+          userId: notification.userId,
+          type: notification.type,
+          payload: { ...notification.payload, ...notificationPayload }
+        }
+
+        await notificationRepository.update(updatedNotification)
+      }
+      return 'Notification updated successfully.'
```

Comment:
same

Reply:
Updated in 77a84e14b8a0b4fe53ac3803ea7b36243c3810f7